application/F3DStarter.cxx

```diff
@@ -798,45 +798,80 @@ int F3DStarter::Start(int argc, char** argv)
     this->Internals->ApplyPositionAndResolution();
 
     f3d::interactor& interactor = this->Internals->Engine->getInteractor();
+
+    interactor.addCommandCallback("load_previous_file_group",
+      [this](const std::string&) { return this->LoadRelativeFileGroup(-1); });
+
+    interactor.addCommandCallback("load_next_file_group",
+      [this](const std::string&) { return this->LoadRelativeFileGroup(+1); });
+
+    interactor.addCommandCallback("reload_current_file_group",
+      [this](const std::string&) { return this->LoadRelativeFileGroup(0, true, true); });
+
+    interactor.addCommandCallback("add_current_directories",
+      [this](const std::string&) -> bool
+      {
+        if (this->Internals->LoadedFiles.size() > 0)
+        {
+          for (const auto& parentPath : F3DInternals::ParentPaths(this->Internals->LoadedFiles))
+          {
+            this->AddFile(parentPath, true);
+          }
+          return this->LoadRelativeFileGroup(0);
+        }
+        return true;
+      });
+
+    interactor.addCommandCallback("take_screenshot",
+      [this](const std::string& filename) -> bool
+      {
+        // TODO add test
+        std::string localFilename =
+          filename.empty() ? this->Internals->AppOptions.ScreenshotFilename : filename;
+        this->SaveScreenshot(localFilename);
+        return true;
+      });
+
+    interactor.addCommandCallback("take_minimal_screenshot",
+      [this](const std::string& filename) -> bool
+      {
+        // TODO add test
+        std::string localFilename =
+          filename.empty() ? this->Internals->AppOptions.ScreenshotFilename : filename;
+        this->SaveScreenshot(localFilename, true);
+        return true;
+      });
+
     interactor.setKeyPressCallBack(
       [this](int, const std::string& keySym) -> bool
       {
         if (keySym == "Left")
         {
-          return this->LoadRelativeFileGroup(-1);
+          return this->Internals->Engine->getInteractor().triggerCommand(
```

Comment:
That'd be great to have a `map<string, string>` and just do something like:
```cpp
interactor.setKeyPressCallBack(
      [this](int, const std::string& keySym) -> bool
      {
        try
        {
          return this->Internals->Engine->getInteractor().triggerCommand(commandMap.at(keySym));
        }
        catch(const std::out_of_range&)
        {
          return false;
        }
      }
);
```

Reply:
this will be reworked in https://github.com/f3d-app/f3d/pull/1687, not needed